internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,314 @@
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeQuotaRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := volumequotarules.NewVolumeQuotaRuleID(subscriptionId, model.ResourceGroupName, model.AccountName, model.CapacityPoolName, model.VolumeName, model.Name)
+			volumeID := volumes.NewVolumeID(subscriptionId, model.ResourceGroupName, model.AccountName, model.CapacityPoolName, model.VolumeName)
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.Get(ctx, id)
+			if err != nil && existing.HttpResponse.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			// Performing some validations that are not possible in the schema
+			if errorList := netAppValidate.ValidateNetAppVolumeQuotaRule(ctx, volumeID, metadata.Client, pointer.To(model)); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			parameters := volumequotarules.VolumeQuotaRule{
+				Location: location.Normalize(model.Location),
+				Properties: &volumequotarules.VolumeQuotaRulesProperties{
+					QuotaSizeInKiBs: pointer.To(model.QuotaSizeInKiB),
+					QuotaType:       pointer.To(volumequotarules.Type(model.QuotaType)),
+					QuotaTarget:     utils.String(model.QuotaTarget),
+				},
+			}
+
+			err = client.CreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume quota rule to be completely provisioned
+			if err := waitForVolumeQuotaRuleCreateOrUpdate(ctx, client, id); err != nil {
+				return err
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeQuotaRules
+
+			id, err := volumequotarules.ParseVolumeQuotaRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Updating %s", id)
+
+			shouldUpdate := false
+			update := volumequotarules.VolumeQuotaRulePatch{
+				Properties: &volumequotarules.VolumeQuotaRulesProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("quota_size_in_kib") {
```

Comment:
Is `quota_size_in_kib` the only updatable attribute? If so, can we throw `ForceNew` on the other attributes. Then we don't have to do the `shouldUpdate` logic as `quota_size_in_kib` will be the only thing updatable.